### DISCUSSION TITLE
JetBrains IDEs (specially Clion) for TinyUSB collabrators

### DISCUSSION THREAD

**Discussion by hathach** [hathach/tinyusb#2306] [Ideas]
JetBrains is kind enough to approve TinyUSB project and provie open source license (all products) for TinyUSB's collaborators https://www.jetbrains.com/community/opensource/#support. 

Collaborators (thank you) that want to try out JetBrains IDE or have any questions please either leave comment and/or drop me an email. Requirement for license approval (by JetBrains) is that you have to consistently committed to tinyusb repo (master) and can only use it with non-commercial opensource projects. Also license is reviewed each year.

FYI: I have been switching to and using Clion for developing TinyUSB for the last year or so and it is much better than any other IDEs I have used previously. Clion has great support for CMake and embedded development and debugging (which is why I have been adding lots of cmake lately). In addition to gcc,clang, Clion also work well with IAR command line as backend.

**Comment by twdr36** [comment]
Hi Hathach,

I am glad I found this thread about the Clion as the IDE for developing TinyUSB.
Could you please share the steps to set up the TinyUSB project into the Clion? I am new to Clion, please forgive my ignorance of it and really appreciate it if you can guide me through setting the TinyUSB project in it.

Thanks.

**Reply by hathach** [reply]
To use with tinyusb example, load the examples/CMakeLists.txt then define your cmake profile with `-DBOARD=yourboard` . For how to use/debug with clion, check out jetbrain documentation

**Comment by patfrench** [comment]
Hi,
I've just bought a CLion licence. Could you do a tutorial on how to start a project with webserial?

I have a feather M0 SAMD21 board from adafruit and I'm having trouble following the getting started.

I also have a problem with my j-link probe: my GDB server sends me back: 
`Truncated register 8 in remote ‘g’ packet` 
I have to ask on the segger forum or I have a problem with my CLion configuration?

Thank you

**Reply by hathach** [reply]
open tinyusb as project, then load the CMake file in `examples/CMakeLists.txt` there is a bunch of cmake profile for boards that I added. You can add your own, for using with JLink, you should checkout jetbrain docs, they are very straightforward

**Reply by patfrench** [reply]
Thank you !
It works well ! **CLion is really good** 
I have one last question: I can't use SEGGER_RTT_Printf

I don't know where to configure it

**Reply by hathach** [reply]
you need to specify `-DLOG=2 -DLOGGER=RTT` in the cmake profile
